google-cloud-storage/test/google/cloud/storage/retry/utils.rb

```diff
@@ -0,0 +1,462 @@
+    policy.bindings.append({"role": role, "members": [member]})
+
+    # IAM policies have no metageneration, clear ETag to avoid checking that it matches.
+    policy.etag = nil unless _preconditions
+    client.set_bucket_policy bucket_name, policy
+  end
+
+  def self.update_hmac_key client, _preconditions, **resources
+    access_id = resources[:hmac_key].access_id
+    etag = resources[:hmac_key].etag
+
+    hmac_key = Google::Apis::StorageV1::HmacKeyMetadata.new access_id: access_id, state: "INACTIVE"
+    hmac_key.etag = etag if _preconditions
+    client.update_hmac_key access_id, hmac_key
+  end
+
+  def self.delete_bucket_acl client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    entity = "allAuthenticatedUsers"
+    client.delete_bucket_acl bucket_name, entity
+  end
+
+  def self.delete_acl client, _preconditions, **resources
+    acl = resources[:bucket].acl
+    acl.delete "allAuthenticatedUsers"
+  end
+
+  def self.insert_bucket_acl client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    entity = CONF_TEST_ACL_ENTITY
+    role = "READER"
+    client.insert_bucket_acl bucket_name, entity, role
+  end
+
+  def self.insert_acl client, _preconditions, **resources
+    acl = resources[:bucket].acl
+    acl.add_writer CONF_TEST_ACL_ENTITY
+  end
+
+  def self.delete_bucket_default_acl client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    entity = "allAuthenticatedUsers"
+    client.delete_default_acl bucket_name, entity
+  end
+
+  def self.delete_default_acl client, _preconditions, **resources
+    default_acl = resources[:bucket].default_acl
+    entity = "allAuthenticatedUsers"
+    default_acl.delete entity
+  end
+
+  def self.insert_bucket_default_acl client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    entity = CONF_TEST_ACL_ENTITY
+    role = "READER"
+    client.insert_default_acl bucket_name, entity, role
+  end
+
+  def self.insert_default_acl client, _preconditions, **resources
+    default_acl = resources[:bucket].default_acl
+    entity = CONF_TEST_ACL_ENTITY
+    default_acl.add_owner entity
+  end
+
+  def self.create_hmac_key client, _preconditions, **resources
+    client.create_hmac_key CONF_TEST_SERVICE_ACCOUNT_EMAIL
+  end
+
+  def self.insert_notification client, preconditions, **resources
+    bucket_name = resources[:bucket].name
+    pubsub_topic_name = CONF_TEST_PUBSUB_TOPIC_NAME
+    client.insert_notification bucket_name, pubsub_topic_name
+  end
+
+  def self.create_notification client, _preconditions, **resources
+    bucket = resources[:bucket]
+    pubsub_topic_name = CONF_TEST_PUBSUB_TOPIC_NAME
+    bucket.create_notification pubsub_topic_name
+  end
+
+  def self.insert_object client, _preconditions, **resources
+    bucket = resources[:bucket]
+    file = StringIO.new CONF_TEST_FILE_CONTENT * 1024 * 1024 # 1MB
```

Comment:
Note this is actually `1024 * 1024 * CONF_TEST_FILE_CONTENT.size` which is 12MB rather than 1MB. Just making sure that's what you intend for this object.